ldms/src/ldmsd/ldmsd_decomp.c

```diff
@@ -156,7 +156,7 @@ static ldmsd_decomp_t __decomp_get(const char *decomp, ldmsd_req_ctxt_t reqc)
 		d = dlopen(library_name, RTLD_NOW);
 		if (d)
 			break;
-		if (!stat(library_name, &st))
+		if (stat(library_name, &st))
```

Comment:
woops. sorry for the bug & thanks for catching it.

Reply:
It seems like this logic could be more helpful to the user. The failing stat() doesn't mean the library does not exist, it could be a permission error; regardless printing the error would be helpful to the user.

If stat() succeeds, we should probably continue after printing the dlerror() and try the next pathdir element, not give up and return NULL.